dashboard/app/models/unit.rb

```diff
@@ -1072,6 +1072,16 @@ def has_other_versions?
     get_course_version&.course_offering&.course_versions&.many?
   end
 
+  # version_year for a given unit should be the same whether you're looking at it on the unit itself,
+  # the associated unit_group (if there is one), or the associated course_version (if there is one).
+  # However, there is at least one instance of unit.version_year being mysteriously inconsistent
+  # with unit.unit_group.version_year and unit.course_version.key for "csp4-2024".
+  # This method is a cautious workaround to reconcile these inconsistencies while we investigate the mismatch.
+  # See https://codedotorg.atlassian.net/browse/TEACH-1492
+  def get_most_recent_version_year
+    [get_course_version&.key.to_i, version_year.to_i, unit_group&.version_year.to_i].max.to_s
+  end
```

Comment:
@davidsbailey what do think about this?

Reply:
I would recommend sticking with `get_course_version&.key` since it is simpler and just as reliable (if not more).

Reply:
oh! when you said "however this still needs to work for units not yet in courses, such as newly created units to be added to a course later" I thought that meant I needed to do something _additional_ to handle newly created units because they aren't added to courses yet and therefore don't yet have an associated `course_version`

Reply:
oh, sorry! I just meant that you had to add the `&.` so that you wouldn't get an error. I don't think units-not-yet-in-courses will have a version_year, anyway.

Reply:
gotcha. thanks for clarifying. I see what you mean now!